app/src/main/java/org/oppia/android/app/home/HomeFragmentPresenter.kt

```diff
@@ -144,7 +150,10 @@ class HomeFragmentPresenter @Inject constructor(
   }
 
   fun onTopicSummaryClicked(topicSummary: TopicSummary) {
-    routeToTopicListener.routeToTopic(internalProfileId, topicSummary.topicId)
+    if (myDownloadsFeatureFlag.get().value)
+      routeToTopicPreviewListener.routeToTopicPreview(internalProfileId, topicSummary.topicId)
+    else
```

Comment:
Could you please add a test to verify that this flow properly opens the preview activity when the activity is enabled? We should also make sure there's already or is now a test for the opposite case (i.e. that the preview activity does not open).